pkg/tbtc/deposit_sweep.go

```diff
@@ -0,0 +1,78 @@
+package tbtc
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+)
+
+// assembleDepositSweepTransaction constructs an unsigned deposit sweep Bitcoin
+// transaction.
+//
+// Regarding input arguments, the walletPublicKey parameter is optional and
+// can be set as nil if the wallet does not have a main UTXO at the moment.
+// The deposits slice must contain at least one element. The fee argument
+// is not validated anyway so must be chosen with respect to the system
+// limitations.
+//
+// The resulting bitcoin.TransactionBuilder instance holds all the data
+// necessary to sign the transaction and obtain a bitcoin.Transaction instance
+// ready to be spread across the Bitcoin network.
+func assembleDepositSweepTransaction(
+	bitcoinChain bitcoin.Chain,
+	walletPublicKey *ecdsa.PublicKey,
+	walletMainUtxo *bitcoin.UnspentTransactionOutput,
+	deposits []*deposit,
+	fee int64,
+) (*bitcoin.TransactionBuilder, error) {
```

Comment:
In the Bridge smart contract we allow the deposit sweep transaction to have a witness (P2WPKH) or non witness (P2PKH) output.
Perhaps we could add a parameter `witness` to this function which could allow choosing the output type?
We did that in our [typescript implementation](https://github.com/keep-network/tbtc-v2/blob/8b9d2629bf4333e650a54f32a4da7cf86bf6785e/typescript/src/deposit-sweep.ts#L119).

Reply:
Yep, I know. The non-witness support in the contracts and typescript was just in case and allowed us to have some flexibility. However, the default choice should be witness as those transactions are usually cheaper. Here we reflect that in the contract code. If for some reason (rather unlikely), we need to support non-witness, we can add it easily in the future.